planner/solver.go

```diff
@@ -537,6 +538,8 @@ func (s *Solver) generatePlacesForSlots(ctx context.Context, req *PlanningReques
 		if len(placesByPrice) == 0 {
 			return nil, fmt.Errorf("failed to find any place for category %s at slot %s for location %+v", slot.Category, slot.TimeSlot.ToString(), req.Location)
 		}
+		// sort places by score descending so the solver checks places with higher score first
+		slices.SortFunc(placesByPrice, func(a, b matching.Place) int { return cmp.Compare(matching.PlaceScore(b), matching.PlaceScore(a)) })
```

Comment:
Is it simple to add some test to check that places are sorted in slices in the solver?

Reply:
We could use [github.com/uber-go/mock](https://github.com/uber-go/mock) to mock `iowrappers.SearchClient` and the `Matcher` interface, it won't be easy though. Let me know if you think we should try it in this PR or in another follow-up.

Reply:
By looking closer at the specific issue, I found that a deeper problem is there should be a "super" place matcher that employ the [strategy pattern](https://refactoring.guru/design-patterns/strategy/go/example), so the solver can just call match once in the super matcher. I'll create a PR to address this.

Reply:
Sg, no need to over-complicate the change in one PR. My original thought was that maybe we can just add a test to verify results from findOptimalSolution are sorted.

Reply:
I forgot to add that it is okay to add this test in a separate PR.